lib/rubocop/cop/performance/intersection_check.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # In Ruby 3.1, `Array#intersect?` has been added.
+      #
+      # This cop identifies places where `(receiver & argument).any?`
+      # can be replaced by `receiver.intersect?(argument)`.
+      #
+      # The `receiver.intersect?(argument)` method is faster than
+      # `(receiver & argument).any?`.
+      #
+      # @example
+      #   # bad
+      #   (receiver & argument).present?
+      #   (receiver & argument).any?
+      #   (receiver & argument).blank?
+      #   (receiver & argument).empty?
+      #
+      #   # good
+      #   receiver.intersect?(argument)
+      #   !receiver.intersect?(argument)
```

Comment:
bad and good examples can be made clearer.

```suggestion
      #   (receiver & argument).present?
      #   (receiver & argument).any?
      #
      #   # good
      #   receiver.intersect?(argument)
      #
      #   # bad
      #   (receiver & argument).blank?
      #   (receiver & argument).empty?
      #
      #   # good
      #   !receiver.intersect?(argument)
```